standard/expressions.md

```diff
@@ -1150,6 +1151,200 @@ object o = (new int[3])[1];
+
+In both cases, the argument list of the call consists of a *format string literal* with *format specifications* for each interpolation, and an argument for each expression corresponding to the format specifications.
+
+The format string literal is constructed as follows, where `N` is the number of interpolations in the *interpolated_string_expression*. The format string literal consists of, in order:
+
+  - The characters of the *Interpolated_Regular_String_Start* or *Interpolated_Verbatim_String_Start*
+  - The characters of the *Interpolated_Regular_String_Mid* or *Interpolated_Verbatim_String_Mid*, if any
+  - Then if `N ≥ 1` for each number `I` from `0` to `N-1`:
+    - A placeholder specification:
+      - A left brace (`{`) character
+      - The decimal representation of `I`
+      - Then, if the corresponding *regular_interpolation* or *verbatim_interpolation* has a *interpolation_minimum_width*, a comma (`,`) followed by the decimal representation of the value of the *constant_expression*
+      - The characters of the *Regular_Interpolation_Format* or *Verbatim_Interpolation_Format*, if any, of the corresponding *regular_interpolation* or *verbatim_interpolation*
+      - A right brace (`}`) character
+    - The characters of the *Interpolated_Regular_String_Mid* or *Interpolated_Verbatim_String_Mid* immediately following the corresponding interpolation, if any
+  - Finally the characters of the *Interpolated_Regular_String_End* or *Interpolated_Verbatim_String_End*.
+
+The subsequent arguments are the *expression*s from the interpolations, if any, in order.
+
+When an *interpolated_string_expression* contains multiple interpolations, the expressions in those interpolations are evaluated in textual order from the left to right.
+
+*Example*:
+
+This example uses the following format specification features:
+
+ - the `X` format specification which formats integers as uppercase hexadecimal,
+ - the default format for a `string` value is the value itself,
+ - positive alignment values that right-justify within the specified field width, and
+ - that `{{` and `}}` are formatted as `{` and `}` respectively.
+
+Given:
+
+```csharp
+string text = "red";
+int number = 14;
+```
+
+Then:
+
+| Interpolated String Expression       | Equivalent Meaning As `string`                                | Value        |
+| :----------------------------------- | :------------------------------------------------------------ | :----------- |
+| `$"{text}"`                          | `string.Format("{0}", text)`                                  | `"red"`      |
+| `$"{{text}}"`                        | `string.Format("{{text}})`                                    | `"{text}"`   |
+| `$"{ text , 4 }"`                    | `string.Format("{0,4}", text)`                                | `" red"`     |
```

Comment:
Is it worth having an example that shows the interpolation_minimum_width being an expression using a named constant rather than just digits?

Reply:
Example of valid code that surprised me:

```csharp
const int Width = 4;
string text = "abc";
string formatted = $"{text, Width}";
```